src/main/java/laba/androidPages/ProductsListPageAndroid.java

```diff
@@ -0,0 +1,139 @@
+package laba.androidPages;
+
+import java.util.*;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+import com.zebrunner.carina.utils.factory.DeviceType;
+import com.zebrunner.carina.utils.mobile.IMobileUtils;
+import com.zebrunner.carina.webdriver.decorator.ExtendedWebElement;
+import com.zebrunner.carina.webdriver.decorator.PageOpeningStrategy;
+import com.zebrunner.carina.webdriver.locator.ExtendedFindBy;
+
+import laba.basePages.ProductsListPage;
+import laba.components.FooterComponent;
+import laba.components.ProductComponent;
+import laba.model.Product;
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = ProductsListPage.class)
+public class ProductsListPageAndroid extends ProductsListPage implements IMobileUtils {
+    @ExtendedFindBy(accessibilityId = "test-PRODUCTS")
+    private ExtendedWebElement title;
+
+    @ExtendedFindBy(accessibilityId = "test-Modal Selector Button")
+    private ExtendedWebElement sortDropdown;
+
+    @FindBy(xpath = "//android.widget.TextView[@text='%s']")
+    private ExtendedWebElement sortOption;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc='test-Item']")
+    private List<ProductComponent> productComponentList;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc='test-Cart']//android.widget.TextView")
+    private ExtendedWebElement cartBadge;
+
+    @FindBy(xpath = "//android.view.ViewGroup[.//android.widget.TextView[contains(@text,'All Rights Reserved')]]")
+    private FooterComponent footerContainer;
+
+    public ProductsListPageAndroid(WebDriver driver) {
+        super(driver);
+        setPageOpeningStrategy(PageOpeningStrategy.BY_ELEMENT);
+        setUiLoadedMarker(title);
+    }
+
+    public FooterComponent getFooter() {
+        return footerContainer;
+    }
+
+    public List<ProductComponent> getProductItems() {
+        return productComponentList;
+    }
+
+    public LoginPageAndroid logout() {
+        getHeaderMenu().openMenu();
+        getHeaderMenu().logoutButtonClick();
+        return new LoginPageAndroid(getDriver());
+    }
+
+    @Override
+    public List<String> getAllProductNames() {
```

Comment:
Duplicate scrolling logic in getAllProductNames(), getAllProductPrices(), and areAllProductNamesVisible(). 
Extract the scroll-and-collect pattern into a generic method.